scipy/stats/_morestats.py

```diff
@@ -4333,16 +4331,21 @@ def median_test(*samples, ties='below', correction=True, lambda_=1,
     return MedianTestResult(stat, p, grand_median, table)
 
 
-def _circfuncs_common(samples, high, low):
+def _circfuncs_common(samples, high, low, xp=None):
+    xp = array_namespace(samples) if xp is None else xp
     # Ensure samples are array-like and size is not zero
-    if samples.size == 0:
-        NaN = _get_nan(samples)
+    if xp_size(samples) == 0:
+        NaN = _get_nan(samples, xp=xp)
```

Comment:
If I understand correctly is this why the branch in `_get_nan` was added?

Reply:
It was one of these. One called it like `_get_nan(*samples)` with `samples` an empty tuple I guess.

Reply:
It was actually in `_axis_nan_policy`. I suppose it *could* happen in other functions that accept a variable number of arguments and the user doesn't pass in any. But the worst that happens is the user gets this error:
```python3
from scipy._lib._array_api import array_namespace
array_namespace()
# TypeError: Unrecognized array input
```
I suppose that could be a little more specific; e.g. "Expected at least one array argument but got none." might be informative enough for `array_namespace` and any function that calls it.